src/main.rs

```diff
@@ -21,22 +21,34 @@ struct Cli {
 
 fn main() {
     let cli = Cli::parse();
-    let locations_file = File::open(&cli.locations)
-        .expect(&format!("Location file not found! (looked in {})", cli.locations.display()));
-    let locations: Vec<Location> = serde_json::from_reader(BufReader::new(locations_file))
-        .expect(&format!("An error occurred while reading the locations file! ({})", cli.locations.display()));
-
-    let regions_file = File::open(&cli.regions)
-        .expect(&format!("Regions file not found! (looked in {})", cli.regions.display()));
-    let mut regions: Vec<Region> = serde_json::from_reader(BufReader::new(regions_file))
-        .expect(&format!("An error occurred while reading the regions file! ({})", cli.regions.display()));
-
-    match_locations_to_regions(&locations, &mut regions);
+    let locations_result: Result<Vec<Location>, std::string::String> = File::open(&cli.locations)
+        .map_err(|e| format!("Location file error! (looked in {}), os: {}", cli.locations.display(), e))
+        .and_then(|locations_file| 
+            serde_json::from_reader(BufReader::new(locations_file))
+                .map_err(|e| format!("An error occurred while reading the locations file! {}", e))
+        )
+        .inspect_err(|e| println!("{}", e));
+    let regions_result: Result<Vec<Region>, std::string::String> = File::open(&cli.regions)
+        .map_err(|e| format!("Region file error! (looked in {}), os: {}", cli.regions.display(), e))
+        .and_then(|regions_file| 
+            serde_json::from_reader(BufReader::new(regions_file))
+                .map_err(|e| format!("An error occurred while reading the regions file! {}", e))
+        )
+        .inspect_err(|e| println!("{}", e));
+
+    let (locations, regions) = match (locations_result, regions_result) {
+        (Ok(locs), Ok(regs)) => (locs, regs),
+        _ => {
+            return;
```

Comment:
I think from an integration point of view this is not a proper thing to do. When an error occurs you should return a non-zero exit code, so that any pipeline that called this tool will also know there was an error. I'd also move the inspect_err here (in fact, push all errors as Types to the edges as much as possible, and then add any side-effecting, e.g. println + return non-zero exit code).